dipy/reconst/dki.py

```diff
@@ -0,0 +1,1634 @@
+        
+            .. math::
+            
+    \begin{matrix} ( & W_{xxxx} & W_{yyyy} & W_{zzzz} & W_{xxxy} & W_{xxxz}
+                     & ... \\
+                     & W_{xyyy} & W_{yyyz} & W_{xzzz} & W_{yzzz} & W_{xxyy}
+                     & ... \\
+                     & W_{xxzz} & W_{yyzz} & W_{xxyz} & W_{xyyz} & W_{xyzz}
+                     & & )\end{matrix}
+
+    Returns
+    -------
+    W : array(4,4,4,4)
+        Full 4D kutosis tensor
+    """
+
+    # Note: The multiplication of the indexes (i+1) * (j+1) * (k+1) * (l+1)
+    # for of an elements is only equal to this multiplication for another
+    # element if an only if the element corresponds to an symmetry element.
+    # This multiplication is therefore used to fill the other elements of the
+    # full kurtosis elements
+    indep_ele = {1: k_elements[0],
+                 16: k_elements[1],
+                 81: k_elements[2],
+                 2: k_elements[3],
+                 3: k_elements[4],
+                 8: k_elements[5],
+                 24: k_elements[6],
+                 27: k_elements[7],
+                 54: k_elements[8],
+                 4: k_elements[9],
+                 9: k_elements[10],
+                 36: k_elements[11],
+                 6: k_elements[12],
+                 12: k_elements[13],
+                 18: k_elements[14]}
+
+    W = np.zeros((3, 3, 3, 3))
+
+    xyz = [0, 1, 2]
+    for ind_i in xyz:
+        for ind_j in xyz:
+            for ind_k in xyz:
+                for ind_l in xyz:
+                    key = (ind_i+1) * (ind_j+1) * (ind_k+1) * (ind_l+1)
+                    W[ind_i][ind_j][ind_k][ind_l] = indep_ele[key]
+
+    return W
+
+
+def split_dki_param(dki_params):
+    r"""
+    Extract the diffusion tensor eigenvalues, the diffusion tensor eigenvector
+    matrix, and the 15 independent elements of the kurtosis tensor from the
+    model parameters estimated from the DKI model
+
+    Parameters
+    ----------
+    dki_params : ndarray (..., 27)
+        All parameters estimated from the diffusion kurtosis model.
+        Parameters are ordered as follow:
+            1) Three diffusion tensor's eingenvalues
+            2) Three lines of the eigenvector matrix each containing the first,
+               second and third coordinates of the eigenvector
+            3) Fifteen elements of the kurtosis tensor
+
+    Returns
+    --------
+    eigvals : array (3,)
+        Eigenvalues from eigen decomposition of the tensor.
+    eigvecs : array (3, 3)
+        Associated eigenvectors from eigen decomposition of the tensor.
+        Eigenvectors are columnar (e.g. eigvecs[:,j] is associated with
+        eigvals[j])
+    kt : array (15,)
+        Fifteen elements of the kurtosis tensor
+    """    
+    evals = dki_params[..., :3]
+    evecs = dki_params[..., 3:12].reshape(dki_params.shape[:-1] + (3, 3))
+    kt = dki_params[..., 12:]
+    
+    return evals, evecs, kt
+
+
+def dki_design_matrix(gtab):
```

Comment:
Because you are already at dki.py module there is no reason repeating dki in front of function names. So, you can use design_matrix here rather than dki_design_matrix.
Now someone would say that that could be a problem if you want to call design_matrix from two different models e.g. dti and dki. In that case you can write
```
from dipy.reconst.dti import design_matrix as dti_design_matrix
from dipy.reconst.dki import design_matrix as dki_design_matrix
```
and resolve any naming issues.